### DISCUSSION TITLE
Better documentation for downloading listener clients

### DISCUSSION THREAD

**Discussion by josephdadams** [josephdadams/TallyArbiter#282] [General]
@JTF4 I think the PR you did recently was a good improvement for the listener clients but I would like to take it a step further.

I think a line like this: 

> Go to https://github.com/josephdadams/TallyArbiter/actions/workflows/build-listener-clients.yaml and click on the first element in the list. From here, scroll down to the "Artifacts" section and click on "TallyArbiter-Listener-M5StickC".

is asking too much of the average user. To those of us on Github, this makes regular and perfect sense. However most users are not on Github all that often (or at all). I get a LOT of emails from users who just couldn't figure it out and that was before we were providing binaries. :)

I'm wondering if there is a more direct way with the documentation to always link to the latest releases of those listener clients/artifact files.

Any ideas?

**Comment by hrueger** [comment]
Hm, I just tried to follow the steps you quoted, and I couldn't even find the artifact section because when I tried to scroll down, my mouse was inside this grey box with all the workflow status reports and it did not scroll down. So I agree, it is quite complicated.

When we add the binaries to GitHub Releases, we can link to them like so: https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases (see the very last section), but that does not seem to work properly.

So I suggest the following: In the Docs, we link to the latest release (https://github.com/josephdadams/TallyArbiter/releases/latest) and then add huge Markdown Buttons to each Release Changelog Text with the Texts "Download for Windows" "Download for Mac", "Download M5 Listener" and so on.

What do you think?

Otherwise as the docs are generated and published with every workflow run anyway, we could change the link in the markdown to a placeholder like `[[LATEST M5 LINK HERE]]` (or something like that) and have the docs action figure out the latest release and replace the placeholder.

**Reply by hrueger** [reply]
Another (although slightly more complicated way) is to create a custom React Component which is used in the docs' download page which fetches the release information via the GitHub API on the client side and then builds up the download links dynamically.

I just did that for the company I work at because we don't always update the docs whenever there's a new release. That's why we need the documentation to always display the latest urls.

**Reply by JTF4** [reply]
I get this all the time. I work in IT and people struggle with the simplest of things. 

That being said, couldn't we just embed the artifacts into the releases? That would make linking easier. 

(Unless that's literally what you suggested and I'm just not completely comprehending it 🙃)

**Reply by josephdadams** [reply]
Yeah if we can get all the various builds added to the latest release I think that would be great. Huge buttons to download even better.
